react/Text/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import styles from './styles.styl'
+
+const mkText = baseClass => props => {
+  const { className, children, tag, ellipsis, ...restProps } = props
+  const Tag = tag
+  return (
+    <Tag className={cx(
+      baseClass, {
+        [styles['u-ellipsis']] : ellipsis
+      },
+      className)}
+      {...restProps}
+    >
+      {children}
+    </Tag>
+  )
+}
```

Comment:
_Note: I copied/pasted a previous component marked as outdated but unresolved._

Am I the only one disturbed by this component?
@GoOz explained to me that he does not want the intermediate `<div />` in case of css layouting. But I don't get the usage:

```jsx
<Text tag="h2">Hello, World!</Text>
```

that result in html to:
```html
<h2 class="g-text">Hello, World!</h2>
```

or

```jsx
<Text>
  <h2>Hello, World!</h2>
</Text>
```

that result in html to:
```html
<div class="g-text">
  <h2>Hello, World!</h2>
</div>
```
---
For that later example, we could get the same html result as the first example with an HOC:
```jsx
const withGText = BaseComponent => props => (
  <BaseComponent {...props} className={cx(props.className, "g-text")} />
```

used like that
```jsx
const HelloWorld = props => <h2>Hello, World!</h2>
const GTextedHelloWorld = withGText(HelloWorld)
```

This results to:
```html
<h2 class="g-text">Hello, World!</h2>
```